### PR TITLE
chore: replace `@ShadowVariable` by `@CascadingUpdateShadowVariable` in some modules

### PR DESCRIPTION
**Pull request by zepfred**
## Description of the change

This PR replaces `@ShadowVariable` with the new shadow variable `@CascadingUpdateShadowVariable` in two modules: `food-packing` and `vehicle-routing`.

There are a total of five modules using custom listeners:
`maintenance-scheduling` - I think there is room for improvement in this particular use case.
`project-job-scheduling` - This module requires a custom listener.
`task-assigning` - Done!
`food-packing` - Done!
`vehicle-routing` - Done!

### Development

- [x] The changes have been covered with tests, if necessary.
- [x] You have a green build, with the exception of the flaky tests.
- [x] UI and JS files are fully tested, the user interface works for all modules affected by your changes (e.g., solve and analyze buttons).
- [x] The network calls work for all modules affected by your changes (e.g., solving a problem).
- [x] The console messages are validated for all modules affected by your changes.

### Code Review

- [x] This pull request includes an explanatory title and description.
- [ ] The GitHub issue is linked.
- [x] At least one other engineer has approved the changes.
- [x] After PR is merged, inform the reporter.

### CODE REVIEW

**Comment by triceo** [comment]
>  task-assigning - I'm unsure if we could use the same approach as in @CascadingUpdateShadowVariable when the source variable resides in another entity.

Interesting! We discussed this situation and decided not to implement it, until we see a use case. Seems like we just found one. :-)

> project-job-scheduling - This module requires a custom listener.

It's good to keep some of the old listeners. They're not deprecated.

File: java/food-packaging/src/main/java/org/acme/foodpackaging/domain/Job.java [hunk 1]
```diff
@@ -40,15 +41,19 @@ public class Job {
...
     private LocalDateTime startCleaningDateTime;
 
-    @PiggybackShadowVariable(shadowVariableName = "startCleaningDateTime")
+    @CascadingUpdateShadowVariable(targetMethodName = "updateStartCleaningDateTime", sourceVariableName = "line")
+    @CascadingUpdateShadowVariable(targetMethodName = "updateStartCleaningDateTime", sourceVariableName = "previousJob")
     private LocalDateTime startProductionDateTime;
 
-    @PiggybackShadowVariable(shadowVariableName = "startCleaningDateTime")
+    @CascadingUpdateShadowVariable(targetMethodName = "updateStartCleaningDateTime", sourceVariableName = "line")
+    @CascadingUpdateShadowVariable(targetMethodName = "updateStartCleaningDateTime", sourceVariableName = "previousJob")
```
Comment (triceo) [inline java/food-packaging/src/main/java/org/acme/foodpackaging/domain/Job.java]: Observation: that's a lot of boilerplate. The annotation - 6 times. The name of the method - 6 times. The name of each variable - three times.

File: java/food-packaging/src/main/java/org/acme/foodpackaging/domain/solver/StartDateTimeUpdatingVariableListener.java [hunk 2]
```diff
@@ -1,102 +0,0 @@
...
-import java.time.LocalDateTime;
-import java.util.Objects;
-
-import org.acme.foodpackaging.domain.Job;
-import org.acme.foodpackaging.domain.Line;
-import org.acme.foodpackaging.domain.PackagingSchedule;
-import ai.timefold.solver.core.api.domain.variable.VariableListener;
-import ai.timefold.solver.core.api.score.director.ScoreDirector;
-
-public class StartDateTimeUpdatingVariableListener implements VariableListener<PackagingSchedule, Job> {
```
Comment (triceo) [inline java/food-packaging/src/main/java/org/acme/foodpackaging/domain/solver/StartDateTimeUpdatingVariableListener.java:12]: Beautiful!

File: java/food-packaging/src/main/java/org/acme/foodpackaging/domain/Job.java (hunk 1 above)
Comment (triceo) [reply java/food-packaging/src/main/java/org/acme/foodpackaging/domain/Job.java]: I now wonder if we should've put the annotation on the method instead. That would completely eliminate all this repetition.

Comment (zepfred) [reply java/food-packaging/src/main/java/org/acme/foodpackaging/domain/Job.java]: Done!

**Comment by zepfred** [comment]
@triceo, I have observed a drop in performance after modifying the quickstart `task-assigning`, and I am currently investigating this issue.

**Review by triceo — APPROVED** [review]

**Comment by triceo** [comment]
CI is failing because it can't find the new code in the solver; that will have to be merged first.

**Comment by zepfred** [comment]
@triceo, let's wait for TimefoldAI/timefold-solver#996 before merging it.

**Pull request merged**